source/endpoint-reference/endpoint-folders.rst

```diff
@@ -0,0 +1,191 @@
+          "size": 10000,
+          "body": {
+              "name": "Folder name"
+          },
+          "index": {
+              "https://app.crunch.io/datasets/abcdef/variables/123/": {
+                  "name": "Birth Year",
+                  "derived": false,
+                  "discarded": false,
+                  "alias": "birthyear",
+                  "type": "numeric",
+                  "id": "123",
+                  "notes": "",
+                  "description": "In what year were you born?"
+              },
+              "https://app.crunch.io/datasets/abcdef/folders/qwe/": {
+                  "name": "Subfolder name"
+              },
+              "https://app.crunch.io/datasets/abcdef/variables/456/": {
+                  "subvariables_catalog": "aad4ad/subvariables/",
+                  "name": "An Array",
+                  "derived": true,
+                  "discarded": false,
+                  "alias": "arrayvar",
+                  "subvariables": [
+                      "439dcf/",
+                      "1c99ea/"
+                  ],
+                  "notes": "All variable types can have notes",
+                  "type": "categorical_array",
+                  "id": "456",
+                  "description": ""
+              }
+          },
+          "graph": [
+              "https://app.crunch.io/datasets/abcdef/variables/456/",
+              "https://app.crunch.io/datasets/abcdef/variables/123/",
+              "https://app.crunch.io/datasets/abcdef/folders/qwe/"
+          ]
+      }
+
+
+Creating folders
+----------------
+
+To create a subfolder, clients must POST to the target folder (which will be the
+parent of the new subfolder).
+
+The payload must contain a `body` member which indicates only the name of
+the subfolder. Note that subfolder names must be unique between them and
+variables included.
+
+Additionally a `graph` member can be included that must contain URLs of the
+varibles that will be children of the new folder. These variables cannot belong
+to another folder (must be ungrouped under root) else the server will return
+ a 400 response.
+To include variables that belong to other folder in the new folder they
+should be moved into the new folder after the fact via PATCH.
+
+.. language_specific::
+   --JSON
+   .. code:: json
+
+      {
+         "entity": "shoji:catalog",
+         "body": {
+            "name": "New subfolder name"
+         },
+         "graph": []
+      }
+
+
+Moving folders and variables
+----------------------------
+
+In order to move folders or variables from one location to another, their URL
+must be included in the `index` catalog sent via a PATCH request to the
+destination folder.
+
+The new elements will be moved out of their existing parents into the new folder
+location.
+
+If the tuple of a folder or variable included on the payload is `null`, then
+that element will be promoted to the root folder.
```

Comment:
It makes sense that this would happen if you PATCH a `null`, but it seems odd to have two ways of doing this (the other being a PATCH of the root). I think we should we simply disallow `null` on these endpoints.

Reply:
Will do https://www.pivotaltracker.com/story/show/154267562